pkg/agent/directive_handler.go

```diff
@@ -90,10 +146,17 @@ func (dh *BasicDirectiveHandler) handleRolloutCommand(ctx context.Context, cmdSc
 	if err != nil {
 		dlog.Errorf(ctx, "error running rollout command %s: %s", cmd, err)
 	}
+
 	dh.reportCommandResult(ctx, commandID, cmd, err, a)
 }
 
-func (dh *BasicDirectiveHandler) reportCommandResult(ctx context.Context, commandID string, cmd *rolloutCommand, cmdError error, a *Agent) {
+type command interface {
+	String() string
+}
+
+func (dh *BasicDirectiveHandler) reportCommandResult(
+	ctx context.Context, commandID string, cmd command, cmdError error, a *Agent,
```

Comment:
Rather than introducing a custom interface, would just using the standard `Stringer` interface be sufficient?

Alternatively, a more heavy handed approach would be to update the protocol buffers to have a "BaseCommand" message type that was extended and then use that. But, not sure that is necessary at this time.